### ISSUE TITLE
Blazor JS to .Net Streaming. IJSStreamReference using same stream reference.

### ISSUE THREAD

**Issue by QadiymStewart** [dotnet/aspnetcore#56912]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When using JS to .Net streaming on second attempt to upload file the files are the same. Looks like the stream reference from JS isn't correctly disposed or being cached and reused. Is this by design? 

### Expected Behavior

On upload,  .Net should get a new stream with the correct data.

### Steps To Reproduce

Repo can be found here https://github.com/QadiymStewart/FileUpload-JSStream

### .NET Version

8.0.303

**Comment by javiercn** [comment]
@QadiymStewart thanks for contacting us.

Thanks for providing a repro, but there seems to be too much going on in it for us to efficiently assess if there is an issue with the sample or the framework.

In these cases, we ask that you reduce the repro to the minimum possible code that reproduces the issue and that you include detailed steps on how to trigger it.

Ideally, this would be a matter of selecting the files and sending them to the server via JS Interop (one question that we have is, why aren't you using InputFile), from 9.0 and on, compression on the websocket is enabled, and won't make a huge difference unless you are passing around very large files.

**Comment by QadiymStewart** [comment]
@javiercn Attempting to do client side compression using browsers native Compression Streams API. 
In the current .NET 8 I don't see any support for FileInput compression.

The repo should be as basic as I can get it just returning the stream from the fileReader . I removed the RCL in the repo. 

Following the example for streaming from JavaScript to .NET
https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-dotnet-from-javascript?view=aspnetcore-8.0#stream-from-javascript-to-net

In the repo the steps are 
1. Page Loads (Home.razor)
2. After render I set the page reference to call back into .NET.
3. User selects file from File Input.
4. User click Upload button 
5. JavaScript Invokes .NET method to get a JavaScript reference steam. (SaveFile())
6. JavaScript reads the file and returns the stream to .NET.
7. .Net save file.

What isn't working. 

When the files from the input are cleared and the user select another file. The original file is being saved again not the newly selected file. Not sure if the stream is being reused or cached when calling "fileDataStream" or it can be that the fileReader isn;t being updated.

 // Get the data stream reference from JavaScript
 var dataReference = await JsRuntime.InvokeAsync<IJSStreamReference>("fileDataStream");

 // Open the stream for reading with a maximum allowed size of 1GB
 await using var dataReferenceStream = await dataReference.OpenReadStreamAsync(maxAllowedSize: 1000000000);
